Chris-L226.en_distribution

cwroney opened the pull request:
Notes: There's a mysterious cost adder in A26globaltech_cost and I have no idea what it represents aside from, broadly, the cost of distribution. Also one grouped input file I commented on (assuming that's supposed to be coming in ungrouped from the other chunk). 

There was a conflict in namespace when I updated from the master branch before committing but I couldn't tell what it was on about so I went ahead with it. 

#634

bpbond commented:
🥇 to @cwroney for amusing commit names

bpbond commented on R/zchunk_L226.en_distribution.R:
```diff
@@ -12,12 +13,10 @@
 #' @importFrom assertthat assert_that
```
`@details` line filled out?

bpbond commented on R/zchunk_L226.en_distribution.R:
```diff
@@ -46,11 +45,16 @@ module_energy_L226.en_distribution_DISABLED <- function(command, ...) {
              "L226.StubTechCoef_gaspipe"))
   } else if(command == driver.MAKE) {
 
+    #Silence global variable package check
```
`# Silence`

bpbond commented on R/zchunk_L226.en_distribution.R:
```diff
@@ -65,172 +69,358 @@ module_energy_L226.en_distribution_DISABLED <- function(command, ...) {
+      repeat_add_columns(tibble("year" = FUTURE_YEARS)) %>%
+      select(-year.x) %>%
+      rename(year = year.y) %>%
+      bind_rows(L226.IO_R_elecownuse_F_Yh, .) ->
```
Why `, .`? Does the order of binding matter?

cwroney replied:
I don't believe so, it just seemed intuitive to me to have the future years come after the base years.

bpbond commented on R/zchunk_L226.en_distribution.R:
```diff
@@ -65,172 +69,358 @@ module_energy_L226.en_distribution_DISABLED <- function(command, ...) {
+      bind_rows(L226.IO_R_elecownuse_F_Yh, .) ->
+      L226.IO_R_elecownuse_F_Y
+
+    #rename columns and round coefficients - L226.StubTechCoef_elecownuse: calibrated coefficients on electricity net ownuse
```
Here and throughout, please ensure a space after `#`

bpbond commented on R/zchunk_L226.en_distribution.R:
```diff
@@ -65,172 +69,358 @@ module_energy_L226.en_distribution_DISABLED <- function(command, ...) {
+
+    # append assumed techchange value and calculate decrease in future energy use
+    L226.IO_R_electd_F_Yfut %>%
+      left_join(A_regions, by = "GCAM_region_ID") %>%
```
Why not `left_join_error_no_match`? Are there region IDs not present in `A_regions`?

cwroney replied:
no, but there's one entry in biomassoil_Tech that's NA and so triggers the error when it copies it (at least I assume that's what's happening).

bpbond commented on inst/extdata/energy/A26.globaltech_cost.csv:
```diff
@@ -0,0 +1,17 @@
+# File: A26.globaltech_cost.csv
+# Title: Final energy delivery cost adders
+# Units: ?
```
Please check with @pkyle or @mwisepnnl or someone about this.

pkyle replied:
In general, these are the energy delivery cost adders, or the difference between producer prices and delivered energy prices to different consumer groups. The unit is 1975$/GJ, and the source for most of the items is the pre-2006 MiniCAM model inputs. Values of zero indicate that delivery costs are included elsewhere in the chain.
A26 also includes some grid reliability (backup) related costs. The backup_electricity cost represents the variable O&M cost of a gas turbine (levelized capital and fixed O&M costs are accounted elsewhere). No such cost is included here for csp_backup, because the turbine costs are already accounted in the primary electric technology costs.

cwroney replied:
Thanks, @pkyle!

bpbond commented:
>Notes: There's a mysterious cost adder in A26globaltech_cost and I have no idea what it represents aside from, broadly, the cost of distribution.

Also - let's clarify this if possible! Thanks.

cwroney commented:
@bpbond There's no XML generating file with this yet. I've never done a level 2 before though -- are we handling the xml code in the same branch as the level 2 code? Certainly the outputs for this would next be used to generate en_distribution.xml.

bpbond commented:
I'll add the xml-generating file. Is this ready for a second review? Thanks.

cwroney commented:
@bpbond it is, it kept throwing merge conflicts in the namespace because the L225 export was removed in the master in the same line L226 was removed here, so I had to manually remove both, but it should be good to go now.

bpbond commented:
This looks good and is just waiting on @mwisepnnl comments/approval.

mwisepnnl commented:
Is the A26 cost adder still mysterious?  It is just as your comment states a cost adder to get from a centralized cost to a distributed cost.  I don't know if the original data system has the reference for these.  If not, maybe raise it as an issue?

cwroney commented:
realized I forgot to add the @details so put them in, also deleted the two misspelled variables that were generating the notes...now redundant with the correctly spelled ones you added. Also, I added the source from Page for the A26 globaltechcost. Done meddling!

bpbond commented:
>Is the A26 cost adder still mysterious? It is just as your comment states a cost adder to get from a centralized cost to a distributed cost. I don't know if the original data system has the reference for these. If not, maybe raise it as an issue?

Good point–thanks. @cwroney please open an issue to follow up on this if you think useful.

bpbond commented:
>now redundant with the correctly spelled ones you added

Haha, I missed that. Anyway thanks for all your work, looks good, merging as soon as tests pass.

cwroney commented:
@bpbond Page said the A26 cost adders were from "pre-2006 MiniCAM model inputs", so I added that as the source. I'm not sure if we need more specific than that, since that's kind of saying that our data source is ourselves in the past?

bpbond commented:
I'm willing to bet that means, practically speaking, "lost in the mists of time". Please do open an issue.
